pandas/io/formats/format.py

```diff
@@ -1621,18 +1621,23 @@ def __init__(
 
     def _format_strings(self) -> list[str]:
         """we by definition have DO NOT have a TZ"""
-        values = self.values
 
+        values = self.values
         if not isinstance(values, DatetimeIndex):
             values = DatetimeIndex(values)
 
         if self.formatter is not None and callable(self.formatter):
-            return [self.formatter(x) for x in values]
+            fmt_values = np.array([self.formatter(x) for x in values])
+        else:
+            fmt_values = values._data._format_native_types(
+                na_rep=self.nat_rep, date_format=self.date_format
+            )
 
-        fmt_values = values._data._format_native_types(
-            na_rep=self.nat_rep, date_format=self.date_format
-        )
-        return fmt_values.tolist()
+        if fmt_values.ndim > 1:
+            nested_formatter = GenericArrayFormatter(fmt_values)
+            return list(nested_formatter.get_result())
+
+        return list(fmt_values)
```

Comment:
why is _anything_ necessary between L1630-L1640?  we know we have DatetimeIndex which is 1D.